### PR TITLE
Improvements and one new event

### PR DESCRIPTION
**Pull request by EchoEllet**
Solve [#1412](https://github.com/singerdmx/flutter-quill/issues/1412)

Some minor changes.

### CODE REVIEW

**Comment by singerdmx** [comment]
@Jon-Salmon please take a look

**Comment by ahmedhnewa** [comment]
I just realized that `gal` is not the same as `gallery_saver`

so some of the changes I made are not necessary

But Gal is still a young library even though it's based on gallery

but I hope you will accept the changes of `afterRemoveImageFromEditor` event
and also new messages "(haven't translated them yet)

File: flutter_quill_extensions/lib/embeds/builders.dart [hunk 1]
```diff
@@ -271,29 +284,39 @@ Widget _menuOptionsForReadonlyImage(
...
+                    // TODO: Please translate this
+                    messenger.showSnackBar(const SnackBar(
+                        content: Text(
+                      'Error while saveing the image',
+                    )));
+                    return;
+                  }
 
-                  // Save image to gallery
-                  await Gal.putImage(imagePath);
```
Comment (Jon-Salmon) [inline flutter_quill_extensions/lib/embeds/builders.dart:292]: `gal` can't be replaced by `image_gallery_saver` as this doesn't support Windows or macOS.

File: flutter_quill_extensions/pubspec.yaml [hunk 2]
```diff
@@ -12,18 +12,21 @@ dependencies:
...
+  image_picker: ">=1.0.4"
   photo_view: ^0.14.0
-  video_player: ^2.7.0
-  youtube_player_flutter: ^8.1.1
-  gal: ^2.1.1
-  math_keyboard: ">=0.1.8 <0.3.0"
-  string_validator: ^1.0.0
-  universal_html: ^2.2.1
-  url_launcher: ^6.1.9
+  video_player: ^2.7.2
```
Comment (Jon-Salmon) [inline flutter_quill_extensions/pubspec.yaml:20]: Please don't bump minimum plugin versions unless there is a major version/breaking change or we need to use a api feature that's only in a newer plugin

File: flutter_quill_extensions/lib/embeds/builders.dart [hunk 3]
```diff
@@ -7,20 +7,23 @@ import 'package:flutter/services.dart';
...
 import 'package:flutter_quill/flutter_quill.dart' hide Text;
 import 'package:flutter_quill/translations.dart';
-import 'package:gal/gal.dart';
-import 'package:http/http.dart' as http;
 import 'package:math_keyboard/math_keyboard.dart';
 import 'package:universal_html/html.dart' as html;
 
 import '../shims/dart_ui_fake.dart'
-    if (dart.library.html) '../shims/dart_ui_real.dart' as ui;
+    if (dart.library.html) 'package:flutter_quill_extensions/shims/dart_ui_real.dart'
```
Comment (Jon-Salmon) [inline flutter_quill_extensions/lib/embeds/builders.dart:14]: Why this change? This goes against the convention of local imports within the package

**Comment by Jon-Salmon** [comment]
@ahmedhnewa I've added some comments to your two merge requests that need addressing before this can be released. Please open a new merge request addressing these.

**Comment by Jon-Salmon** [comment]
@singerdmx Please can you give me a bit more time before merging things in which affects the extensions package? I'm only able to review code on this repository 1-2 times a week.

**Comment by singerdmx** [comment]
ok, sorry about it

**Comment by ahmedhnewa** [comment]
> ok, sorry about it

Please give me time I fill fix everything

File: flutter_quill_extensions/lib/embeds/builders.dart (hunk 3 above)
Comment (ahmedhnewa) [reply flutter_quill_extensions/lib/embeds/builders.dart:14]: Because when I was making the changes I didn't fork it I was not planning on sending pull requests, I just made a custom version of this file so I have to change stuff but the IDE does not notify me about the import if it against the convention of local imports of the package, I have changed it now
